src/expr/src/scalar/mod.rs

```diff
@@ -884,6 +884,44 @@ impl MirScalarExpr {
                                     })
                                     .collect(),
                             }
+                        } else if let (
+                            BinaryFunc::Eq,
+                            literal @ MirScalarExpr::Literal(Ok(..), ..),
+                            MirScalarExpr::CallUnary { func, expr },
+                        ) = (&*func, &**expr1, &**expr2)
+                        {
+                            // <lit> = f(<expr>), where f is invertible
```

Comment:
nit: it's a bit easier to follow if you use consistent naming of meta-variables in the docs and the Rust variables in the actual code. For example here the Rust variables are `literal`, `func`, `expr`, but in the code they are `lit`, `f`, `expr`.

Reply:
fixed